packages/http/test/index.js

```diff
@@ -509,13 +509,54 @@ describe('post', () => {
       return body;
     });
 
+    testServer
+      .post('/api/csv-reader')
+      .times(3)
+      .reply(200, function (url, body) {
+        console.log('thanks for the message!', body);
+        return body;
+      });
+
     testServer
       .post('/api/fake-custom-success-codes')
       .reply(302, function (url, body) {
         return { ...body, statusCode: 302 };
       });
   });
 
+  it('should make an http request from inside the parseCSV callback', async function () {
+    const csv = 'id,name\n1,taylor\n2,mtuchi\n3,joe\n4,stu\n5,elias';
+    const state = { references: [], data: [], apiResponses: [] };
+
+    const resultingState = await parseCsv(
+      csv,
+      { chunkSize: 2 },
+      async (state, rows) => {
+        // TODO: how do we write these in the real world? Like this?
+        if (rows.length > 0)
```

Comment:
If you're looking for a more functional solution, you could declare the callback like this:

```
(state, rows) => rows.length && post(...)
```
You wouldn't need to async/await the post because it'll return a promise.

Reply:
@josephjclark i tried this approach 
```
(state, rows) =>
  rows.length > 0 &&
  post(
    'https://www.example.com/api/csv-reader',
    {
      body: rows,
    },
    state => {
      console.log("I'm a regular callback");
      return state;
    }
  )
``` 
It did not work. i was getting this error 
```
 AssertionError: expected [Function] to deeply equal { data: { references: [], …(2) } }
 ```
 
 Then tried 
 ```
 (state, rows) =>
  rows.length > 0 &&
  post(
    'https://www.example.com/api/csv-reader',
    {
      body: rows,
    },
    state => {
      console.log("I'm a regular callback");
      return state;
    }
  )(state)
  ```
  Now i am getting this error 
  ```
   TypeError: state.references is not iterable
   ```
   I think we should pair to see pair to discuss this further, You can see the [current implementation of parseCsv here ](https://github.com/OpenFn/adaptors/blob/7034947eded80f6dc69681be3ec61a7dd38dbcd8/packages/common/src/Adaptor.js#L565)